tests/functional/tests/rgw/test_rgws.py

```diff
@@ -0,0 +1,30 @@
+import pytest
+
+
+class TestRGWs(object):
+
+    @pytest.mark.no_docker
+    def test_ceph_rgw_package_is_installed(self, node, Package):
+        # ceph-radosgw is the package name on centos, radosgw name on ubuntu
+        assert (Package("ceph-radosgw").is_installed or Package("radosgw").is_installed)
+
+
+    def test_rgw_services_are_running(self, node, Service):
+        service_name = "ceph-radosgw@rgw.ceph-{hostname}".format(
+            hostname=node["vars"]["inventory_hostname"]
+        )
+        assert Service(service_name).is_running
+
+    def test_rgw_services_are_enabled(self, node, Service):
+        service_name = "ceph-radosgw@rgw.ceph-{hostname}".format(
+            hostname=node["vars"]["inventory_hostname"]
+        )
+        assert Service(service_name).is_enabled
+
+    @pytest.mark.no_docker
+    def test_rgw_http_endpoint(self, node, Interface, Command):
+        # rgw frontends ip_addr is configured on eth0
+        ip_addr = Interface("eth0").addresses[0]
+        cmd = "curl http://{ip_addr}:{port}".format(ip_addr=ip_addr, port=8080)
```

Comment:
Can you use ``Socket`` instead of ``curl`` here? https://testinfra.readthedocs.io/en/latest/modules.html#socket